.github/workflows/build.yml

```diff
@@ -0,0 +1,52 @@
+name: "Linting and Testing"
+
+on:
+  push:
+    branches: [master,github-actions-test]
+    paths-ignore:
+    - 'README.md'
+    - '.deepsource.toml'
+
+  pull_request:
+    branches: [master]
+    paths-ignore:
+    - 'README.md'
+    - '.deepsource.toml'
+
+
+jobs:
+  build:
+
+    runs-on: ${{matrix.os}}
+    strategy:
+      matrix:
+        os: [ubuntu-18.04]
+        python-version: [3.7,3.8,3.9]
+        test-folder : ['tests']
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build using Python ${{matrix.python-version}}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{matrix.python-version}}
+
+      - name: cache conda
+        uses: actions/cache@v1
+        with:
+          path: $CONDA
+          key: ${{ runner.os }}-conda-${{ hashFiles('simSPI.yml') }}
+          restore-keys: |
+            ${{ runner.os }}-conda-
+      - name: install dependencies
+        run:  |
+          # $CONDA is an environment variable pointing to the root of the miniconda directory
+          $CONDA/bin/conda env update --file simSPI.yml --name base
```

Comment:
@geoffwoollard you can install ioSPI as a dependenci in simSPI.yml using pip.

Reply:
So under `dependencies` in simSPI.yml I'm adding `git+https://github.com/comSPI/ioSPI.git`, where something like `numpy` would usually go. This will need me to put in `fourier.py` into that, and make sure it installs (add in dependencies, etc)... I'll get another PR going for that.